tests/Unit/Domain/CoordinateMaps/Test_UniformCylindricalSide.cpp

```diff
@@ -0,0 +1,517 @@
+             radius_two * sin(theta_max_minus_two))
+            ? std::numeric_limits<double>::lowest()
+            : -(radius_two * sin(theta_max_minus_two) -
+                radius_one * sin(theta_max_minus_one)) *
+                  tan(min_alpha_minus);
+    CHECK(min_center_one_z_from_alpha_minus <=
+          max_center_one_z_from_alpha_plus);
+    // max_center_one_z comes from the restriction
+    // z_plane_plus_two >= z_plane_plus_one + 0.03*radius_two,
+    // and the restriction
+    // 0.98 r_2 >= r_1 + | C_1 - C_2 |
+    // and the restriction
+    // C^z_1 < C^z_2 + r_1 + r_2/5
+    // which is designed to not allow a tiny sphere 1 at the edge of
+    // a large sphere 2.
+    const double max_center_one_z =
+        std::min({max_center_one_z_from_alpha_plus,
+                  z_plane_plus_two - z_plane_frac_plus_one * radius_one -
+                      0.03 * radius_two,
+                  center_two[2] + radius_one + 0.2 * radius_two,
+                  center_two[2] + 0.98 * radius_two - radius_one});
+    // min_center_one_z comes from the restriction
+    // z_plane_minus_two <= z_plane_minus_one - 0.03*radius_two,
+    // and the restriction
+    // 0.98 r_2 >= r_1 + |C_1 - C_2 |
+    // and the restriction
+    // C^z_1 > C^z_2 - r_1 - r_2/5
+    // which is designed to not allow a tiny sphere 1 at the edge of
+    // a large sphere 2.
+    const double min_center_one_z =
+        std::max({min_center_one_z_from_alpha_minus,
+                  z_plane_minus_two - z_plane_frac_minus_one * radius_one +
+                      0.03 * radius_two,
+                  center_two[2] - radius_one - 0.2 * radius_two,
+                  center_two[2] - 0.98 * radius_two + radius_one});
+    CHECK(min_center_one_z <= max_center_one_z);
+    return min_center_one_z +
+           unit_dis(gen) * (max_center_one_z - min_center_one_z);
+  }();
+  CAPTURE(center_one_z);
+
+  // Now we can compute z_plane_plus_one and z_plane_minus_one If
+  // upper and lower planes are equal, put in exact value so there is
+  // no roundoff.
+  const double z_plane_plus_one =
+      center_one_z + radius_one * z_plane_frac_plus_one;
+  CAPTURE(z_plane_plus_one);
+  const double z_plane_minus_one =
+      center_one_z + radius_one * z_plane_frac_minus_one;
+  CAPTURE(z_plane_minus_one);
+
+  // Only thing remaining are the x and y centers of sphere_one.
+  const double horizontal_distance_spheres =
+      [&z_plane_plus_one, &z_plane_plus_two, &theta_max_plus_one,
+       &theta_max_plus_two, &min_alpha_plus, &z_plane_minus_one,
+       &z_plane_minus_two, &theta_max_minus_one, &theta_max_minus_two,
+       &min_alpha_minus, &center_one_z, &center_two, &radius_one, &radius_two,
+       &unit_dis, &gen]() {
+        // Let rho be the horizontal (x-y) distance between the centers of
+        // the spheres.
+
+        // maximum rho so that sphere2 is inside of sphere 1, with a
+        // safety factor of 0.98
```

Comment:
Again, will the map fail if this safety factor isn't there?

Reply:
I reworded the comment.  This safety factor isn't new; its the same restriction that  `0.98 R_2 <= R_1 + |C_1-C_2|` but that places a limit on rho now that the radii and z-coords of the centers are now known.